packages/@glimmer/local-debug-flags/index.ts

```diff
@@ -1,23 +1,171 @@
-export const LOCAL_DEBUG: true | false =
-  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-  // @ts-ignore
-  import.meta.env.VM_LOCAL_DEV &&
-  (() => {
-    let location = typeof window !== 'undefined' && window.location;
-    if (location && /[&?]disable_local_debug/u.test(window.location.search)) {
-      return false;
+/* eslint-disable no-console */
+
+declare global {
+  interface ImportMetaEnv {
+    VM_LOCAL_DEV?: boolean;
+  }
+}
+
+// All of these flags are expected to become constant `false` in production builds.
```

Comment:
And those that exist through to ember (via debug mode, which is always used) should be strippable through ember's build process.
And those that make it through that (hopefully just DEBUG at that point) should be strippable by userland terser